cropper/src/main/java/com/canhub/cropper/CropImageContract.kt

```diff
@@ -0,0 +1,489 @@
+package com.canhub.cropper
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.Rect
+import android.net.Uri
+import android.os.Bundle
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.annotation.DrawableRes
+import com.canhub.cropper.CropImage.getActivityResult
+import com.canhub.cropper.CropImageView.CropShape
+import com.canhub.cropper.CropImageView.Guidelines
+import com.canhub.cropper.CropImageView.RequestSizeOptions
+
+class CropImageContract :
+    ActivityResultContract<CropImageContractOptions, CropImageView.CropResult>() {
+
+    override fun createIntent(context: Context, input: CropImageContractOptions): Intent {
+        input.options.validate()
+        return Intent().apply {
+            val bundle = Bundle()
+            bundle.putParcelable(CropImage.CROP_IMAGE_EXTRA_SOURCE, input.uri)
+            bundle.putParcelable(CropImage.CROP_IMAGE_EXTRA_OPTIONS, input.options)
+            putExtra(CropImage.CROP_IMAGE_EXTRA_BUNDLE, bundle)
+        }
+    }
+
+    override fun parseResult(
+        resultCode: Int,
+        intent: Intent?
+    ): CropImageView.CropResult {
+        val result = getActivityResult(intent)
+        return if (result == null || resultCode == Activity.RESULT_CANCELED) {
+            CropImage.CancelledResult
+        } else {
+            result
+        }
+    }
+}
+
+fun options(
+    uri: Uri? = null,
+    builder: CropImageContractOptions.() -> (Unit) = {}
+): CropImageContractOptions {
+    val options = CropImageContractOptions(uri, CropImageOptions())
+    options.run(builder)
+    return options
+}
+
+data class CropImageContractOptions(
```

Comment:
What you think about this be another file? I know is directly connect to the contract, but maybe this will improve the files reading.